src/components/Statistics/WeeklyReport.js

```diff
@@ -0,0 +1,93 @@
+import { Dataset, LtePlusMobiledataOutlined, PlusOneOutlined } from '@mui/icons-material';
+import * as React from 'react';
+import { useState, useEffect, useMemo } from 'react';
+import {Bar} from 'react-chartjs-2';
+import { useNavigate } from "react-router";
+import  {Chart as ChartJS} from 'chart.js/auto'
+import axios from 'axios';
+
+export  function WeeklyStatistic(){
+
+function diff_minutes(dt2, dt1)
+{
+var diff =(new Date(dt2).getTime() - new Date(dt1).getTime()) / 1000;
+return Math.abs(Math.round(diff));
+}
+
+const [courses, setCourses] = useState([])
+  
+  useEffect(async()=>{
+    
+   async function fetchData(){
+   var username = localStorage.getItem("username")
+   username = username.substring(1);
+   username= username.slice(0, -1)
+   
+     await axios.get(`${process.env.REACT_APP_BASE_URL}events/study-events-weekly/${username}`)
+   .then((response) =>{
+ 
+     console.log(response.data)
+     var arr= response.data
+     var courseList=[]           
+      arr.forEach((event)=>{ 
+       console.log(event)      
+      var courseName = event.subject + event.catalog
+       
+       if (courseList.some(e => e.name === courseName)) {
```

Comment:
duplicate code from monthly report. Please create a util file with code that are usefull to both components.